Formula/node.rb

```diff
@@ -56,30 +56,11 @@ def install
 
     if build.with? "npm"
       resource("npm").stage buildpath/"npm_install"
-
-      # make sure npm can find node
-      ENV.prepend_path "PATH", bin
```

Comment:
If you think this is grim, for testing my current local changes to the `brew` side are this:

```diff
diff --git a/Library/Homebrew/language/node.rb b/Library/Homebrew/language/node.rb
index ab2206a5b..7bcd5c2f9 100644
--- a/Library/Homebrew/language/node.rb
+++ b/Library/Homebrew/language/node.rb
@@ -15,6 +15,12 @@ module Language
       # explicitly use our npm and node-gyp executables instead of the user
       # managed ones in HOMEBREW_PREFIX/lib/node_modules which might be broken
       ENV.prepend_path "PATH", Formula["node"].opt_libexec/"npm/bin"
+      # npm 5.0.0 pretty much smashed the assumption Homebrew made that npm
+      # would always generate installable directories out-of-tree.
+      # We can restore the old behaviour that made package management
+      # relatively possible by packing the path up first. Obviously, this is
+      # not ideal.
+      @filename = Utils.popen_read("npm pack")
     end
 
     def self.std_npm_install_args(libexec)
@@ -23,7 +29,7 @@ module Language
       # (or creating a new one if no .npmignore file already exists)
       open(".npmignore", "a") { |f| f.write("\n.brew_home\n") }
       # npm install args for global style module format installed into libexec
-      ["--verbose", "--global", "--prefix=#{libexec}", "."]
+      ["--verbose", "--global", "--prefix=#{libexec}", "./#{@filename}"]
     end
 
     def self.local_npm_install_args
```